[master < T350] Add graphml docs (#1020)
memgraph/docs: mage/query-modules/python/import-util.md

</Tabs>

### `graphml(path, config)`

#### Input:

* `path: string` ➡ path to the graphML file that is being imported.
* `config: Map (default={})` ➡ configuration parameters explained below.

#### Parameters:

| Name 	             | Type   | Default	| Description 	                                                |
|-	                 |-	      |-	      |-	                                                            |

> | readLabels          | Bool 	| False	  | Create node labels based on the value of the property whose key is "labels". 	|

Reviewer: ```suggestion
| readLabels          | Bool 	| False	  | Create node labels by using the value of the `labels` property. |
```

1. nodes ain't people
2. the original sentence is very complicated... is mine ok?

| defaultRelationshipType  | String	| "RELATED"  	  | The default relationship type to use when none is specified in the import file. 	|
| storeNodeIds 	         | Bool 	| False   	| Store node's id attribute as a property.	|

> | source           | Map 	| { }	  | A map that has two keys: label and id. Label is mandatory while id's default value is "id". This allows the import of relationships in case the source node is not present in the file. It will search for a source node with a specific label and a property whose key is this map's id value, and whose value is relationship's source node id. For example, with a config map `{source: {id: 'serial_number', label: 'Device'}}` and an edge defined as `<edge id="e0" source="n0" target="n1" label="CONNECT"><data key="label">CONNECT</data></edge>`, if node "n0" doesn't exist, it will search for a source node `(:Device {serial_number: "n0"})`.  |

Reviewer: ```suggestion
| source           | Map 	| { }	  | A map with two keys: `label` and `id`. The `label` is mandatory, while the `id`'s default value is `id`. This allows the import of relationships if the source node is absent in the file. It will search for a source node with a specific label and a property equal to the map's ID value. The value of that property should be equal to the relationship's source node ID. For example, with a config map `{source: {id: 'serial_number', label: 'Device'}}` and an edge defined as `<edge id="e0" source="n0" target="n1" label="CONNECT"><data key="label">CONNECT</data></edge>`, if node "n0" doesn't exist, it will search for a source node `(:Device {serial_number: "n0"})`.  |
```
What will allow the import of relationships if the source node is absent?
What will search for a source node?

The explanation was really complicated, is what I wrote true?

> | target     | Map 	| { }	  | A map that has two keys: label and id. Label is mandatory while id's default value is "id". This allows the import of relationships in case the target node is not present in the file. It will search for a target node with a specific label and a property whose key is this map's id value, and whose value is relationship's target node id. For example, with a config map `{target: {id: 'serial_number', label: 'Device'}}` and an edge defined as `<edge id="e0" source="n0" target="n1" label="CONNECT"><data key="label">CONNECT</data></edge>`, if node "n1" doesn't exist, it will search for a target node `(:Device {serial_number: "n1"})`.  	|

Reviewer: ```suggestion
| target     | Map 	| { }	  | A map with two keys: `label` and `id`.  The `label` is mandatory while the `id`'s default value is `id`. This allows the import of relationships in case the target node is absent in the file. It will search for a target node with a specific label and a property equal to the map's `id` value. The value of that property should be equal to the relationship's target node ID. For example, with a config map `{target: {id: 'serial_number', label: 'Device'}}` and an edge defined as `<edge id="e0" source="n0" target="n1" label="CONNECT"><data key="label">CONNECT</data></edge>`, if node "n1" doesn't exist, it will search for a target node `(:Device {serial_number: "n1"})`.  	|
```

#### Output:

> * `status: string` ➡ "success" if no errors are generated.

Reviewer: ```suggestion
* `status: string` ➡ `success` if no errors are generated.
```

Reviewer: why is there an asterisk at the beginning of this row'?

#### Usage:

The `path` you have to provide as procedure argument depends on how you started
Memgraph.

<Tabs
  groupId="export_to_json_usage"
  defaultValue="docker"
  values={[
    {label: 'Docker', value: 'docker'},
    {label: 'Linux', value: 'linux'},
  ]
}> 

<TabItem value="docker">

If you ran Memgraph with Docker, database will be exported to a graphML file inside
the Docker container. We recommend exporting the database to the graphML file
inside the `/usr/lib/memgraph/query_modules` directory.

You can call the procedure by running the following query:

```cypher
CALL export_util.graphML(path);
```

> where `path` is the path to the JSON file inside the

Reviewer: ```suggestion
where `path` is the path to the JSON file inside the
```

CSV or JSON?

`/usr/lib/memgraph/query_modules` directory in the running Docker container (e.g.,
`/usr/lib/memgraph/query_modules/export.graphml`).

:::info
You can [**copy the exported CSV file to your local file system**](/memgraph/how-to-guides/work-with-docker#how-to-copy-files-from-and-to-a-docker-container) using the [`docker cp`](https://docs.docker.com/engine/reference/commandline/cp/) command.
:::
</TabItem>

<TabItem value="linux">

To export database to a local graphML file create a new directory (for example,
`export_folder`) and run the following command to give the user `memgraph` the
necessary permissions:

```
sudo chown memgraph export_folder
```

Then, call the procedure by running the following query:

```cypher
CALL export_util.graphml(path);
```

> where `path` is the path to a local JSON file that will be created inside the

Reviewer: ```suggestion
where `path` is the path to a local JSON file that will be created inside the
```

JSON or CSV?
